lib/serve/functions.js

```diff
@@ -0,0 +1,103 @@
+'use strict';
+
+var _ = require('lodash');
+var chalk = require('chalk');
+var path = require('path');
+var RSVP = require('rsvp');
+var getProjectId = require('../getProjectId');
+var logger = require('../logger');
+var utils = require('../utils');
+var parseTriggers = require('../parseTriggers');
+var EmulatorController;
+var controller;
+
+function _pollOperation(service, operation) {
+  return new RSVP.Promise(function(resolve, reject) {
+    var poll = function() {
+      service.operations.get({
+        name: operation[0].name
+      }, function(err, _operation) {
+        if (err) {
+          reject(err);
+        } else if (_operation.done) {
+        // TODO: handle deploy failures
+          resolve(_operation.response.value);
+        } else {
+          setTimeout(poll, 500);
+        }
+      });
+    };
+    poll();
+  });
+}
+
+function _startEmulator(options) {
+  try {
+    EmulatorController = require('@google-cloud/functions-emulator/src/cli/controller');
+  } catch (err) {
+    var msg = err;
+    if (process.version !== 'v6.9.1') {
+      msg = 'Please use Node version v6.9.1, you have ' + process.version + '\n';
+    }
+    utils.logWarning('Cannot start functions emulator. ' + msg);
+    return RSVP.reject();
+  }
+  if (process.version !== 'v6.9.1') {
+    utils.logWarning('The functions emulator works best with Node version v6.9.1, you have ' + process.version + '\n');
+  }
+  var functionsDir = path.join(options.config.projectDir, options.config.get('functions.source'));
+  controller = new EmulatorController({verbose: true});
+  // TODO: pipe logs to console
+  logger.info(chalk.bold('Functions Directory:'), options.config.get('functions.source'), '\n');
+
+  return controller.start().then(function() {
```

Comment:
What port number is used here? Is it configurable? I think ideally we probably want it to work like:

Hosting served on `--port`, Functions on `--port + 1`